contracts/augur/trading/ZeroXTrade.sol

```diff
@@ -1,178 +1,35 @@
-    /// @param owner  The address of the token holder
-    /// @param id     ID of the Token
-    /// @return       The _owner's balance of the Token type requested
-    function balanceOf(address owner, uint256 id) external view returns (uint256) {
-        (address _market, uint256 _price, uint8 _outcome, uint8 _type) = unpackTokenId(id);
-        Order.Types _orderType = Order.Types(_type);
-        if (_orderType == Order.Types.Ask) {
-            return askBalance(owner, IMarket(_market), _outcome, _price);
-        } else if (_orderType == Order.Types.Bid) {
-            return bidBalance(owner, IMarket(_market), _outcome, _price);
-        }
-    }
-
-    function totalSupply(uint256 id) external view returns (uint256) {
-        return 0;
-    }
-
-    function bidBalance(address _owner, IMarket _market, uint8 _outcome, uint256 _price) public view returns (uint256) {
-        uint256 _numberOfOutcomes = _market.getNumberOfOutcomes();
-        // Figure out how many almost-complete-sets (just missing `outcome` share) the creator has
-        uint256[] memory _shortOutcomes = new uint256[](_numberOfOutcomes - 1);
-        uint256 _indexOutcome = 0;
-        for (uint256 _i = 0; _i < _numberOfOutcomes - 1; _i++) {
-            if (_i == _outcome) {
-                _indexOutcome++;
-            }
-            _shortOutcomes[_i] = _indexOutcome;
-            _indexOutcome++;
-        }
-
-        uint256 _attoSharesOwned = shareToken.lowestBalanceOfMarketOutcomes(_market, _shortOutcomes, _owner);
-
-        uint256 _attoSharesPurchasable = cash.balanceOf(_owner).div(_price);
-
-        return _attoSharesOwned.add(_attoSharesPurchasable);
-    }
-
-    function askBalance(address _owner, IMarket _market, uint8 _outcome, uint256 _price) public view returns (uint256) {
-        uint256 _attoSharesOwned = shareToken.balanceOfMarketOutcome(_market, _outcome, _owner);
-        uint256 _attoSharesPurchasable = cash.balanceOf(_owner).div(_market.getNumTicks().sub(_price));
-
-        return _attoSharesOwned.add(_attoSharesPurchasable);
-    }
-
-    /// @notice Get the balance of multiple account/token pairs
-    /// @param owners The addresses of the token holders
-    /// @param ids    ID of the Tokens
-    /// @return        The _owner's balance of the Token types requested
-    function balanceOfBatch(address[] calldata owners, uint256[] calldata ids) external view returns (uint256[] memory balances_) {
-        balances_ = new uint256[](owners.length);
-        for (uint256 _i = 0; _i < owners.length; _i++) {
-            balances_[_i] = this.balanceOf(owners[_i], ids[_i]);
-        }
-    }
-
-    function setApprovalForAll(address operator, bool approved) external {
-        revert("Not supported");
-    }
-
-    function isApprovedForAll(address owner, address operator) external view returns (bool) {
-        return true;
-    }
+    // @Discuss Not sure why zeroXTrade requires another ERC1155. How is it different from ShareToken?
```

Comment:
This contract fulfills an ERC1155 interface so that it can be specified as the token being traded in 0x signed orders. The actual trade data is packed into the tokenId provided (contains the marketId, price, outcome, and direction) and the makerTokenAssetData (contains the kycToken and exchange address).